### PR TITLE
AG-12969 - Exit annotation text edit mode on zoom change

### PR DESCRIPTION
**Pull request by manapeirov**
PR for https://ag-grid.atlassian.net/browse/AG-12969

### CODE REVIEW

**Review by alantreadway — APPROVED** [review]

File: packages/ag-charts-enterprise/src/features/annotations/annotationsStateMachine.ts [hunk 1]
```diff
@@ -338,6 +350,8 @@ export class AnnotationsStateMachine extends StateMachine<States, AnnotationType
                     },
                 ],
 
+                zoomChange: actionReset,
```
Comment (lsjroberts) [inline packages/ag-charts-enterprise/src/features/annotations/annotationsStateMachine.ts]: I wouldn't have thought we'd want to deselect the annotation on zoom?

**Review by lsjroberts — APPROVED** [review]
LGTM, one small question.

File: packages/ag-charts-enterprise/src/features/annotations/annotationsStateMachine.ts (hunk 1 above)
Comment (manapeirov) [reply packages/ag-charts-enterprise/src/features/annotations/annotationsStateMachine.ts]: Not sure actually maybe we can ask David, I imagine if it goes out of view we wouldn't want it to be selected

Comment (manapeirov) [reply packages/ag-charts-enterprise/src/features/annotations/annotationsStateMachine.ts]: Updated to undo

**Review by alantreadway — APPROVED** [review]
LGTM

**Pull request merged**